lib/tasks/event.rake

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+namespace :events do
+  INITIAL_STATE_EVENTS = %w[GenericEvent::MoveProposed GenericEvent::MoveRequested].freeze
+  INITIAL_STATUSES = %(proposed requested)
+
+  desc 'Backfills MoveProposed and MoveRequested events since start of time'
+  task add_missing_initial_state_events: :environment do
+    dry_run = ENV.fetch('DRY_RUN', 'true') == 'true'
+
+    puts "DRY_RUN: #{dry_run}"
+    puts
+
+    report = {
+      moves_with_initial_state_events: 0,
+      moves_without_initial_state_events: 0,
+      moves_with_invalid_initial_state: 0,
+      proposed_events_to_be_created: 0,
+      requested_events_to_be_created: 0,
+      events_created: 0,
+    }
+    GenericEvent.record_timestamps = false
+    GenericEvent.no_touching do
+      Move.all.find_each do |move|
```

Comment:
is `.all` necessary here? in older versions of rails that would instantiate every move in the database into memory